File: src/types/question.ts
// 问题类型枚举（与后端对应）
export enum QuestionType {
  Single = 0,
  Judge = 1,
  Fill = 2,
  Program = 3,
  ShortAnswer = 4
}

// 题型映射类型
export interface TypeMap {
  [key: string]: string;
  [QuestionType.Single]: string;
  [QuestionType.Judge]: string;
  [QuestionType.Fill]: string;
  [QuestionType.Program]: string;
  [QuestionType.ShortAnswer]: string;
}

// 搜索参数接口
export interface SearchParams {
  keyword?: string;
  type?: QuestionType;
  dateRange?: [Date, Date];
}

// 基础问题接口
export interface BaseQuestion {
  id?: number;
  createTime?: string;
  question: string;
  analysis: string;
}

// 单选题表单接口
export interface SingleQuestionForm extends BaseQuestion {
  type: QuestionType.Single;
  options: string[];
  correctAnswer: string;
}

// 判断题表单接口
export interface JudgeQuestionForm extends BaseQuestion {
  type: QuestionType.Judge;
  correctAnswer: '正确' | '错误';
}

// 填空题表单接口
export interface FillQuestionForm extends BaseQuestion {
  type: QuestionType.Fill;
  answers: string[];
}

// 编程题表单接口
export interface ProgramQuestionForm extends BaseQuestion {
  type: QuestionType.Program;
  sampleInput: string;
  sampleOutput: string;
  referenceAnswer: string;
}

// 简答题表单接口
export interface ShortAnswerQuestionForm extends BaseQuestion {
  type: QuestionType.ShortAnswer;
  referenceAnswer: string;
}

// 题目 DTO 类型联合
export type QuestionForm =
  | SingleQuestionForm
  | JudgeQuestionForm
  | FillQuestionForm
  | ProgramQuestionForm
  | ShortAnswerQuestionForm;

// 表格数据接口
export interface TableQuestion {
  id: number;
  type: QuestionType;
  question: string;
  options?: string[];
  correctAnswer?: string;
  answers?: string[];
  sampleInput?: string;
  sampleOutput?: string;
  referenceAnswer?: string;
  analysis?: string;
  createTime: string;
  category?: string;
  tags?: string[];
}
export interface ApiQuestion {
  id: number;
  type: QuestionType;
  content: string;
  optionsJson?: string;
  answersJson?: string;
  examplesJson?: string;
  tagsJson?: string;
  analysis?: string;
  referenceAnswer?: string;
  createTime?: string;
  difficulty?: number;
  category?: string;
}

File: QuestionBankApi/Models/Question.cs
/*
 * Question.cs
 * ------------
 * 题目实体模型，对应数据库中的题目表结构。
 * 定义了题目的各种属性，包括内容、选项、答案、解析、示例、分类、难度等。
 */

using QuestionBankApi.Enums;

namespace QuestionBankApi.Models;

/// <summary>
/// 题目实体类，映射到数据库中的题目表
/// </summary>
public class Question
{
    /// <summary>
    /// 题目ID，主键
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// 题目类型（枚举）
    /// </summary>
    public QuestionType Type { get; set; }

    /// <summary>
    /// 题目内容
    /// </summary>
    public string Content { get; set; } = null!;

    /// <summary>
    /// 题目选项，JSON格式字符串
    /// </summary>
    public string? OptionsJson { get; set; }

    /// <summary>
    /// 答案，JSON格式字符串
    /// </summary>
    public string? AnswersJson { get; set; }

    /// <summary>
    /// 题目解析
    /// </summary>
    public string? Analysis { get; set; }

    /// <summary>
    /// 示例数据，JSON格式字符串
    /// </summary>
    public string? ExamplesJson { get; set; }

    /// <summary>
    /// 参考答案
    /// </summary>
    public string? ReferenceAnswer { get; set; }

    /// <summary>
    /// 题目分类
    /// </summary>
    public string? Category { get; set; }

    /// <summary>
    /// 难度等级
    /// </summary>
    public int Difficulty { get; set; }

    /// <summary>
    /// 标签，JSON格式字符串
    /// </summary>
    public string? TagsJson { get; set; }

    /// <summary>
    /// 创建时间
    /// </summary>
    public DateTime CreateTime { get; set; }
}
